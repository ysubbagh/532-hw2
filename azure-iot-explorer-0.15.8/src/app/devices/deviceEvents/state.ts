/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import { Type } from 'protobufjs';
import { Message } from '../../api/models/messages';

export type FormMode = 'fetched' | 'initialized' | 'upserted' | 'working' | 'failed' | 'updating' | 'setDecoderSucceeded' | 'setDecoderFailed';

export type DecodeType = 'JSON' | 'Protobuf';

export interface ContentTypeState {
    decodeType: DecodeType;
    decoderProtoFile?: File;
    decoderPrototype?: Type;
}

export interface DeviceEventsStateInterface {
    message: Message[];
    contentType: ContentTypeState;
    formMode: FormMode;
}

export const getInitialDeviceEventsState = (): DeviceEventsStateInterface => ({
    contentType: {
        decodeType: 'JSON'
    },
    formMode: 'initialized',
    message: []
});
