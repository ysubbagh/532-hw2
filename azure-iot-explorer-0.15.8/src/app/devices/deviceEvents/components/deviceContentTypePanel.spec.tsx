/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import 'jest';
import * as React from 'react';
import { mount, shallow } from 'enzyme';
import { DeviceContentTypePanel } from './deviceContentTypePanel';
import * as deviceEventsStateContext from '../context/deviceEventsStateContext';
import { getInitialDeviceEventsState } from '../state';

describe('DeviceDecoderPanel', () => {
    it('matches snapshot with default json option', () => {
        jest.spyOn(deviceEventsStateContext, 'useDeviceEventsStateContext').mockReturnValue(
            [getInitialDeviceEventsState(), deviceEventsStateContext.getInitialDeviceEventsOps()]);
        expect(shallow(
            <DeviceContentTypePanel
                showContentTypePanel={true}
                onToggleContentTypePanel={jest.fn()}
            />)).toMatchSnapshot();
    });

    it('matches snapshot with customize content type option', () => {
        jest.spyOn(deviceEventsStateContext, 'useDeviceEventsStateContext').mockReturnValue(
            [{...getInitialDeviceEventsState(), contentType: {decodeType: 'Protobuf'}},
                deviceEventsStateContext.getInitialDeviceEventsOps()]);
        expect(shallow(
            <DeviceContentTypePanel
                showContentTypePanel={true}
                onToggleContentTypePanel={jest.fn()}
            />)).toMatchSnapshot();
    });

    it('expect setDecoderInfo called when save is clicked with correct input', () => {
        const setDecoderInfo = jest.fn();
        jest.spyOn(deviceEventsStateContext, 'useDeviceEventsStateContext').mockReturnValue(
            [{...getInitialDeviceEventsState(), contentType: {decodeType: 'Protobuf', decoderProtoFile: new File([], '')}},
                {...deviceEventsStateContext.getInitialDeviceEventsOps(), setDecoderInfo}]);
        const wrapper = mount(
            <DeviceContentTypePanel
                showContentTypePanel={true}
                onToggleContentTypePanel={jest.fn()}
            />);
        const form = wrapper.find('form');
        form.simulate('submit');
        wrapper.update();
        expect(setDecoderInfo).toBeCalled();
    });
});
 