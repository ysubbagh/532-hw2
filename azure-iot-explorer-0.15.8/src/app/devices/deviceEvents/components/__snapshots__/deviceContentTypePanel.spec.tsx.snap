// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceDecoderPanel matches snapshot with customize content type option 1`] = `
<StyledPanelBase
  closeButtonAriaLabel="common.close"
  headerText="deviceEvents.customizeContentType.header"
  isBlocking={false}
  isOpen={true}
  onDismiss={[MockFunction]}
  type={3}
>
  <form
    onSubmit={[Function]}
  >
    <Stack
      tokens={
        Object {
          "childrenGap": 10,
        }
      }
    >
      <Dropdown
        defaultSelectedKey="Protobuf"
        disabled={false}
        label="deviceEvents.customizeContentType.contentTypeOption.label"
        onChange={[Function]}
        options={
          Array [
            Object {
              "key": "JSON",
              "text": "JSON",
            },
            Object {
              "key": "Protobuf",
              "text": "Protobuf",
            },
          ]
        }
        required={true}
      />
      <StyledLabelBase>
        deviceEvents.customizeContentType.protobuf.file.label
      </StyledLabelBase>
      <input
        accept=".proto"
        disabled={false}
        id="protoFile"
        name="protoFile"
        onChange={[Function]}
        required={true}
        type="file"
      />
      <StyledTextFieldBase
        disabled={false}
        label="deviceEvents.customizeContentType.protobuf.type.label"
        onChange={[Function]}
        placeholder="deviceEvents.customizeContentType.protobuf.type.placeholder"
        readOnly={false}
      />
      <CustomizedPrimaryButton
        ariaLabel="deviceEvents.customizeContentType.save"
        disabled={false}
        text="deviceEvents.customizeContentType.save"
        type="submit"
      />
    </Stack>
  </form>
</StyledPanelBase>
`;

exports[`DeviceDecoderPanel matches snapshot with default json option 1`] = `
<StyledPanelBase
  closeButtonAriaLabel="common.close"
  headerText="deviceEvents.customizeContentType.header"
  isBlocking={false}
  isOpen={true}
  onDismiss={[MockFunction]}
  type={3}
>
  <form
    onSubmit={[Function]}
  >
    <Stack
      tokens={
        Object {
          "childrenGap": 10,
        }
      }
    >
      <Dropdown
        defaultSelectedKey="JSON"
        disabled={false}
        label="deviceEvents.customizeContentType.contentTypeOption.label"
        onChange={[Function]}
        options={
          Array [
            Object {
              "key": "JSON",
              "text": "JSON",
            },
            Object {
              "key": "Protobuf",
              "text": "Protobuf",
            },
          ]
        }
        required={true}
      />
      <CustomizedPrimaryButton
        ariaLabel="deviceEvents.customizeContentType.save"
        disabled={false}
        text="deviceEvents.customizeContentType.save"
        type="submit"
      />
    </Stack>
  </form>
</StyledPanelBase>
`;
